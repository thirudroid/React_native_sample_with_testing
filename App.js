import React from 'react'
import {
  TouchableHighlight,
  Text,
  View,
  StyleSheet
} from 'react-native'

import { connect } from 'react-redux'
import { fetchPeopleFromAPI} from './actions'

let styles

const App = (props) => {
  const{
    container,
    text,
    button,
    buttonText
  } = styles

  const { people, isFetching} = props.people

  return (
    <View style={container}>
      <Text style={text}>Reduc App</Text>
      <TouchableHighlight onPress={props.getPeople} style={button}>
        <Text style={buttonText}>Fetch Data</Text>
      </TouchableHighlight>

      {
        isFetching && <Text>Loading...</Text>
      }{
        people.length ? (
            people.map((persion,index) => {
              return(
                <View key={index}>
                  <Text>Name: {persion.name}</Text>
                </View>
              )
            })
        ):null
      }
    </View>
  )
}


styles = StyleSheet.create({
  container:{
    marginTop: 100,
    paddingLeft: 20,
    paddingRight: 20
  },
  text:{
    textAlign: 'center'
  },
  button:{
    backgroundColor: '#0b7eff',
    height: 60,
    justifyContent: 'center',
    alignItems: 'center'
  },
  buttonText:{
    color: 'white'
  }
})

function mapStateToProps(state){
  return{
    people: state.people
  }
}

function mapDispatchToProps(dispatch){
  return{
    getPeople:() => dispatch(fetchPeopleFromAPI())
  }
}

export default connect(mapStateToProps,mapDispatchToProps)(App)
