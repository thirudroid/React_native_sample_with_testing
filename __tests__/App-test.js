/**
 * @format
 */

import 'react-native';
import {TouchableHighlight,Text} from 'react-native';
import renderer from 'react-test-renderer'
import React from 'react';
import App from '../App';
import configureStore from '../configureStore';
import {Provider} from 'react-redux';
import {
  FETCHING_PEOPLE,
  FETCHING_PEOPLE_SUCCESS,
  FETCHING_PEOPLE_FAILURE
} from '../constants'
import peopleReducer from '../reducers/people'

const store = configureStore()
const snap = renderer.create(<Provider store = {store}>
  <App/>
  </Provider>);

it('renders correctly', () => {
  expect(snap.toJSON()).toMatchSnapshot();
  const button = snap.root.findByType(TouchableHighlight);
  const text = button.findByType(Text);
  expect(text.props.children).toBe("Fetch Data");
});

describe('Reducers', ()=> {
  let initialState = {
    people: [],
    isFetching: false,
    error: false
}

it('Should be initialState', ()=> {
  let reducer = peopleReducer(undefined, {});
  expect(reducer).toEqual(initialState);
 });

 it('should handle FETCH_PEOPLE', () => {
  expect(
    peopleReducer(initialState, {
      type: FETCHING_PEOPLE
    })
  ).toEqual(
    {"error": false, "isFetching": true, "people": []}
  )})

  it('should handle FETCH_PEOPLE_SUCCESS', () => {
    expect(
      peopleReducer(initialState, {
        type: FETCHING_PEOPLE_SUCCESS
      })
    ).toEqual(
      {"error": false, "isFetching": false}
    )})

    it('should handle FETCH_PEOPLE_FAILURE', () => {
      expect(
        peopleReducer(initialState, {
          type: FETCHING_PEOPLE_FAILURE
        })
      ).toEqual(
        {"error": true, "isFetching": false,"people":[]}
      )})

});