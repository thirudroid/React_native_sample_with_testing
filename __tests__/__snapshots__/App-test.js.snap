// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 100,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Reduc App
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0b7eff",
        "height": 60,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Fetch Data
    </Text>
  </View>
</View>
`;
